targets/XPlatCppSdk/source/cppsdk/source/playfab/PlayFabHttp.cpp

```diff
@@ -226,20 +229,23 @@ namespace PlayFab
             throw std::runtime_error("You should not call Update() when PlayFabSettings::threadedCallbacks == true");
 
         CallRequestContainer* reqContainer;
-        size_t resultCount;
-
         { // LOCK httpRequestMutex
             std::unique_lock<std::mutex> lock(httpRequestMutex);
-            resultCount = pendingResults.size();
-            if (resultCount == 0)
-                return resultCount;
+            if (pendingResults.empty())
+                return activeRequestCount;
 
             reqContainer = pendingResults[pendingResults.size() - 1];
             pendingResults.pop_back();
+            activeRequestCount--;
         } // UNLOCK httpRequestMutex
 
         HandleResults(*reqContainer);
         delete reqContainer;
-        return resultCount;
+
+        // activeRequestCount can be altered by HandleResults, so we have to re-lock and return an updated value
```

Comment:
I'm not seeing how HandleResults can modify activeRequestsCount - can you please help me understand ?

Reply:
HandleResults invokes customer code.
Customer code can do anything, including start another API call.

Reply:
agreed